src/controller/CHIPDeviceController.cpp

```diff
@@ -528,6 +533,52 @@ void DeviceController::OnValue(const char * key, const char * value) {}
 
 void DeviceController::OnStatus(const char * key, Operation op, CHIP_ERROR err) {}
 
+#if CHIP_ENABLE_MDNS
```

Comment:
Could we avoid the use of #ifdef by using some generic interface?

if no mdns is used, there seems to be no harm to have the method implemented and it also looks like we may want to enable the same functionality using DNS-SD in thread (which may not use mDNS)

Reply:
The Resolver is such a generic interface, so I would need to always build `lib/mdns/Resolver_ImplNone.cpp`. Note that although Thread devices will not use mDNS-SD to register services, the controller will still use mDNS-SD. It's just that the border router will serve as a proxy for Thread devices. Therefore, there's probably no need to introduce yet another layer.

Reply:
Ok, I changed it so that we always build mdns code, but if `chip_enable_mdns=false`, the None implementations are used.